core/services/relay/evm/functions/logpoller_wrapper.go

```diff
@@ -419,10 +420,27 @@ func (l *logPollerWrapper) handleRouteUpdate(ctx context.Context, activeCoordina
 			l.lggr.Errorw("LogPollerWrapper: Failed to update routes", "err", err)
 		}
 	}
+
+	filters := l.logPoller.GetFilters()
+	for _, filter := range filters {
+		if filter.Name[:len(l.filterPrefix())] != l.filterPrefix() {
+			continue
+		}
+		if filter.Name == l.filterName(l.activeCoordinator) || filter.Name == l.filterName(l.proposedCoordinator) {
+			continue
+		}
+		if err := l.logPoller.UnregisterFilter(ctx, filter.Name); err != nil {
+			l.lggr.Errorw("LogPollerWrapper: Failed to unregister filter", "filterName", filter.Name, "err", err)
+		}
```

Comment:
nit: could also info-log for successful unregister

Reply:
I think an info log is a bit higher level than necessary. I am adding a debug log tho.